src/raftstore/store/peer.rs

```diff
@@ -1696,8 +1696,20 @@ impl Peer {
             return Ok((resp, None));
         }
 
+        let compact_term = req.get_compact_log().get_compact_term();
+        if compact_term == 0 {
+            info!("{} compact term missing in {:?}, skip.",
```

Comment:
I need to write 100+ lines to cover this, and this is only for backward compatibility. Once TiKV is upgraded this will not be visited again. Maybe I add a TODO here first, and add the unit tests later to cover not only here but also any other message integrity check.